### DISCUSSION TITLE
NetCDF Climatology Files

### DISCUSSION THREAD

**Discussion by jir245** [dtcenter/METplus#1469] [File I/O] [answered]
I'm running ensemble_stat and trying to pass climatology data for skill score stats.  I'm using NetCDF data files for both the climatology mean and standard deviation information.  I'm getting the following messages:
WARNING: read_climo_file() -> Trouble reading climatology file 

The climatology files include lat/lon information as well as mean and standard deviation fields.  Is there additional information that needs to be in the netcdf files.

Thanks for your help.

**Comment by CPKalb** [comment]
Hi:

It would be helpful to see how the data is stored in the climatology files.  Would you be able to upload a file so that we can take a look?  Information on how to send us data can be found here: https://github.com/dtcenter/METplus/discussions/954

MET can read different types of netCDF files (CF-compliant, WRF pinterp, MET version netCDF), but it is difficult to know what may be missing without seeing the file.

Thanks,
Christina

**Reply by jir245** [reply]
Hi Christina,

I put a netcdf file on ftp.

Thanks for your help.

**Reply by CPKalb** [reply]
Thanks.  What directory is this data in?  I couldn't tell what your surname was from your Github name.

Christina

**Comment by jir245** [comment]
Oh, sorry.  It's under Rubin_data.

Thanks,
Juli

**Comment by CPKalb** [comment, marked as answer]
Hi:

Taking a look at your climatology file, it isn't actually in a format that MET can easily read.  However, it is close to a CF compliant format.  I was able to get plot_data_plane to read this file by telling MET to read it as if it were CF compliant.  To do that, I added the following, which should go after the variable level; file_type = NETCDF_NCCF;
So, the climatology field would look something like this:
name=[{AOD_mean"; level="( * , * )"; file_type=NETCDF_NCCF;}]

Alternatively, you could add information to your netCDF file to make it CF compliant.  To make it CF complaint, you would definitely need a time variable, avariable called grid_mapping_name and a global attribute that tells which CF conventions are used.  This would be something like Convetions = "CF-1.8" or whichever version of CF your file follows.  (https://cfconventions.org/Data/cf-conventions/cf-conventions-1.9/cf-conventions.html).

Picture from plot_data_plane:
![testAOD](https://user-images.githubusercontent.com/22180338/156829480-a03fb19b-667c-406b-9919-456b276bf88a.png)


Do either of these work in ensemble_stat for you?

Christina

**Comment by jir245** [comment]
Thanks,  I will adjust the netcdf file information like you mentioned and
see if I can get it to work.  I'll let you know if I have any issues.

Juli

**Reply by CPKalb** [reply]
Hi Juli:

I'm checking in to see if you were able to get MET to work on the climatology files.

Thanks,
Christina

**Comment by CPKalb** [comment]
Hi Juli:

Since we haven't heard more, I’m going to go ahead and mark this one as answered and lock the Discussion.  Please feel free to start a new Discussion with any other issues you encounter.

Thanks,
Christina
